src/tests/nativeaot/SmokeTests/UnitTests/Interfaces.cs

```diff
@@ -689,6 +689,8 @@ class C3<T> : IInterface, IInterface<T>
         static IInterface<object> s_c3a = new C3<object>();
         static IInterface s_c3b = new C3<object>();
 
+        // Works around https://github.com/dotnet/runtime/pull/94342
+        [MethodImpl(MethodImplOptions.NoOptimization)]
```

Comment:
This found a new bug. The `Run` method is essentially optimized to a throw. It's ridiculous how well this optimizes:

* We figured out the value assigned to the static is read only using whole program view (the static is not marked read only in source code)
* RyuJIT then figured out exact type information from the preinit data blob.
* RyuJIT then devirtualized the call (into the wrong method, but that's the bug)
* RyuJIT then folded the string constant comparison.

All that's left is the throw (and a couple garbage instructions I couldn't fully make sense of).